### PR TITLE
🪟 🎉 Adds a TextWithOverflowTooltip component

### PR DESCRIPTION
**Pull request by tealjulia**
## What
**TL;DR**: Uses `react-resize-detector` and `react-intersection-observer` to decide whether to wrap a Text component in a Tooltip.

## How
- Check if the HTMLElement inside the Text component is in view (using `react-intersection-observer`)
- If it is, compare the scroll width to the client width to determine if the text is overflowed
- If it is, wrap the element in a tooltip
- Re-checks these things if the element is resized (using `react-resize-detector`)

## Recommended reading order
1. `TextWithOverflowTooltip.*`
2. `CatalogTreeTableCell.tsx`
3. `CatalogTreeTableRow.tsx`
4. others

## Can this PR be safely reverted / rolled back?
*If you know that your PR is backwards-compatible and can be simply reverted or rolled back, check the YES box.*

*Otherwise if your PR has a breaking change, like a database migration for example, check the NO box.*

*If unsure, leave it blank.*
- [X] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-webapp/src/components/ui/Text/Text.tsx [hunk 1]
```diff
@@ -42,29 +43,36 @@ const getTextClassNames = ({
...
-    color = "default",
-    gradient = false,
-    ...remainingProps
-  }) => {
-    const className = classNames(
-      getTextClassNames({ centered, size, bold, inverseColor, gradient, color }),
-      classNameProp
-    );
+export const Text: React.FC<React.PropsWithRef<React.PropsWithChildren<TextProps>>> = React.memo(
+  React.forwardRef(
```
Comment (tealjulia) [inline airbyte-webapp/src/components/ui/Text/Text.tsx:47]: Ugly diff -- this was wrapped in a forwardRef and the ref was passed down to the text

File: airbyte-webapp/src/components/ui/PillSelect/PillButton.tsx [hunk 2]
```diff
@@ -47,9 +49,15 @@ export const PillButton: React.FC<PillButtonProps> = ({
...
-            </Text>
+            {withOverflowTooltip ? (
+              <TextWithOverflowTooltip as="span" size="xs" className={styles.text}>
+                {child}
+              </TextWithOverflowTooltip>
+            ) : (
+              <Text as="span" size="xs" className={styles.text}>
+                {child}
+              </Text>
+            )}
```
Comment (krishnaglick) [inline airbyte-webapp/src/components/ui/PillSelect/PillButton.tsx:60]: I think this is may be the wrong level for the tooltip.
My PK selection cell won't show overflow unless I select 2 or more items.

File: airbyte-webapp/src/components/ui/Text/TextWithOverflowTooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+              inViewRef(element);
+            }}
+            className={classNames(className, styles.overflowText)}
+            {...props}
+          >
+            {children}
+          </Text>
+        }
+        disabled={tooltipDisabled}
+        className={styles.fullWidthTooltip} // todo: is that right?
```
Comment (krishnaglick) [inline airbyte-webapp/src/components/ui/Text/TextWithOverflowTooltip.tsx:51]: I'm unsure what logic you're porting from but this will allow the tooltip to be any width.

Comment (tealjulia) [reply airbyte-webapp/src/components/ui/Text/TextWithOverflowTooltip.tsx:51]: I pulled this in from somewhere in the catalog tree table row.

Looking at this with fresh eyes... We probably _don't_ want this by default so the tooltip will wrap.

Comment (krishnaglick) [reply airbyte-webapp/src/components/ui/Text/TextWithOverflowTooltip.tsx:51]: I'd agree with that. I have a tickling in the back of my head that I applied some width-adjusting style in my table work. Lemme know if you'd like to pair to sort through this!

Comment (tealjulia) [reply airbyte-webapp/src/components/ui/Text/TextWithOverflowTooltip.tsx]: It was for when things have very long strings with no word break ie:


added a `break-word` instead, which seems cleaner?:

**Comment by tealjulia** [comment]
migrated

File: airbyte-webapp/src/components/ui/PillSelect/PillButton.tsx (hunk 2 above)
Comment (tealjulia) [reply airbyte-webapp/src/components/ui/PillSelect/PillButton.tsx:60]: D'oh!  It's the `as="span"`.  Can't overflow text in a span :)
